File: 3345. Smallest Divisible Digit Product I/Solution.py
class Solution:
    def smallestNumber(self, n: int, t: int) -> int:
        def digit_product(num):
            product = 1
            for digit in str(num):
                product *= int(digit)
            return product
        current = n
        while True:
            if digit_product(current) % t == 0:
                return current
            current += 1

File: 3345. Smallest Divisible Digit Product I/Solution.js
/**
 * @param {number} n
 * @param {number} t
 * @return {number}
 */
var smallestNumber = function(n, t) {
    function digitProduct(num) {
        let product = 1;
        const str = String(num);
        for (let i = 0; i < str.length; i++) {
            product *= parseInt(str[i]);
        }
        return product;
    }
    
    let current = n;
    while (true) {
        if (digitProduct(current) % t === 0) {
            return current;
        }
        current++;
    }
};
